src/places/models/place.model.ts

```diff
@@ -83,6 +89,47 @@ export class Place extends Model {
 
       return newestDemand.updatedAt;
     },
+
+    set() {
+      throw new ForbiddenOperationError('Cannot set virtual field lastUpdatedAt');
+    },
   })
   lastUpdatedAt!: Date | null;
+
+  @ApiProperty({
+    nullable: false,
+    type: 'number',
+    description:
+      'virtual field that returns priority of place calculated from demands with priorities we have the most of',
+  })
+  @Column({
+    type: DataType.VIRTUAL,
+    get() {
+      const demands: Demand[] = this.getDataValue('demands');
+
+      if (!demands || !demands.length) {
+        return 0;
+      }
+
+      const countedPrios = demands.reduce((counted: Record<number, number>, demand: Demand): Record<number, number> => {
```

Comment:
It took a while since I understood what is going on in the code 😅 

Maybe we could describe it more?

also this code will only work when we have only to priorities.

```js
      const importanceToCountMap = demands.reduce((importanceToCount: Record<number, number>, demand: Demand): Record<number, number> => {
        if (demand.priority) {
          importanceToCount[demand.priority.importance]
            ? importanceToCount[demand.priority.importance]++
            : (importanceToCount[demand.priority.importance] = 1);
        }

        return importanceToCount;
      }, {});

     // INFO: find the most common importance tuple
      const mostCommonImportance =  Object.keys(importanceToCountMap).reduce(
        (mostCommon: { key: 0, count: 0 }, currentKey: number) => {
           if (mostCommon.count <= importanceToCountMap[currentKey]) {
              return { key: currentKey, importanceToCountMap[currentKey]]
           }

           return mostCommon;
        },
        { key: 0, count: 0 },
      );

    return mostCommonImportance.key;
    },
```